File: CodeHelper/CodeHelper.Core/Parse/ParseResults/ViewModels/Statements/Declare.cs
using System;
using System.Collections.Generic;
using CodeHelper.Core.Parse.ParseResults.ViewModels.Expressions;
namespace CodeHelper.Core.Parse.ParseResults.ViewModels.Statements
{
    public class Declare : TokenPair
    {
        public String Type_name { get; set; }
        public String Name { get; set; }
        public Expression Default_value { get; set; }

        public VariableInfo Parse()
        {
            //var rslt = new FieldInfo();
            //rslt.Name = this.Name;
            //rslt.Type_name = this.Type_name;
            //return rslt;

            var rslt = new VariableInfo(this);
            rslt.Name = this.Name;
            rslt.Type_name = this.Type_name;
            if (this.Default_value != null)
            {
                rslt.Value = Default_value.Parse();
            }
            return rslt;
        }
    }
}

File: ToolBag/ParseTool/src/parsetool/viewmodel/models/statements/Declare.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package parsetool.viewmodel.models.statements;

import parsetool.models.common.TokenPair;
import parsetool.viewmodel.models.expressions.Expression;

/**
 *
 * @author cqy
 */
public class Declare extends TokenPair{
    private String type_name;
    private String name;
    private Expression default_value;

    /**
     * @return the type_name
     */
    public String getType_name() {
        return type_name;
    }

    /**
     * @param type_name the type_name to set
     */
    public void setType_name(String type_name) {
        this.type_name = type_name;
    }

    /**
     * @return the name
     */
    public String getName() {
        return name;
    }

    /**
     * @param name the name to set
     */
    public void setName(String name) {
        this.name = name;
    }

    /**
     * @return the default_value
     */
    public Expression getDefault_value() {
        return default_value;
    }

    /**
     * @param default_value the default_value to set
     */
    public void setDefault_value(Expression default_value) {
        this.default_value = default_value;
    }
}
